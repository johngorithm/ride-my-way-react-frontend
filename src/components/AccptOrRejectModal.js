import React, { Fragment } from 'react';
import Button from 'components/Atom/Button';

class AcceptOrRejectModal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <Fragment>
        <div className="modal" id="reject-ride-request-modal">
          <div className="modal-content animate">
            <div className="tile">
              <div className="tile-body center-text">
                <p className="confirm-msg small">
                  Are you sure you want to <strong>ACCEPT</strong> a ride request
                  from <span>John</span>
                </p>
                <p className="error-message smaller" />
              </div>

              <div className="tile-footer center-text">
                <Button 
                  classes="button button-blue close"
                  text="CLOSE" 
                />
                <Button
                  // onClick="acceptOrRejectRequest(this)"
                  classes="button button-white yes-btn"
                
                  text = "YES"
                />
              </div>
            </div>
          </div>
        </div>
      </Fragment>
    );
  }
}

AcceptOrRejectModal.propTypes = {};

export default AcceptOrRejectModal;
