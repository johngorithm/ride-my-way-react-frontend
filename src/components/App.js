import React, { Component } from 'react';
import '../syles/App.css';

class App extends Component {
    render () {
        return (
            <div>
                <h1>Good Morning Dera</h1>
            </div>
        );
    }
}

export default App;