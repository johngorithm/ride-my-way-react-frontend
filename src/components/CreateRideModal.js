import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router';

import InputGroup from 'components/Atom/InputGroup';
import Button from 'components/Atom/Button';

import { closeModal } from 'actions/modalActions';
import { createRide } from 'actions/rideActions';
import validateRide from 'utils/validateRide'

export class CreateRideModal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      rideInfo: {
        destination: '',
        date: '',
        time: '',
        takeOffVenue: '',
        capacity: ''
      },
      errors: {
        destination: '',
        date: '',
        time: '',
        takeOffVenue: '',
        capacity: ''
      }
    };
  }

  handleChange = (e) => {
    const { rideInfo } = this.state;
    const newRideData = { ...rideInfo };

    newRideData[`${e.target.name}`] = e.target.value;

    this.setState({
      rideInfo: newRideData
    });
  }

  closeModal(e) {
    e.preventDefault();
    this.props.closeModal(this.props.modalName)
  }

  handleSubmit = event => {
    event.preventDefault();
    const { rideInfo } = this.state;

    const { errors, isValidData } = validateRide(rideInfo);

    if (isValidData) {
      this.setState({
        errors: {
          destination: '',
          date: '',
          time: '',
          takeOffVenue: '',
          capacity: ''
        }
      })
      return this.props.createRide(rideInfo).then(() => {
        this.props.closeModal(this.props.modalName)
        this.props.history.push('/home');
      })
    } else {
      this.setState({
        errors
      });
    }
  }

  render() {
    const { destination, date, time, capacity, takeOffVenue } = this.state.rideInfo;
    const { errors } = this.state;
    const { succesMessge, isLoading } = this.props;
    return (
      <Fragment>
        <div className="modal" id="add-offer-modal">
          <div className="modal-content animate">
            <form id="create-offer-form" onSubmit={this.handleSubmit}>
              <div id="form-header" className="co-xl-12 co-lg-12 co-md-12 co-sm-12" >
                  <h3 id="form-header-title" className ="center-text">OFFER A RIDE</h3>
                  {succesMessge ? (<p className="success-message small center-text">{succesMessge}</p>) : null}
                  
              </div>
              <div className="row">
                <InputGroup  
                  classes="co-xl-12 co-lg-12 co-md-12 co-sm-12"
                  label="Destination"
                  error={errors.destination}
                  type="text"
                  name="destination"
                  placeholder="where are you goin?"
                  value={destination}
                  onChange={this.handleChange}
                />

                <InputGroup  
                  classes="co-xl-6 co-lg-6 co-md-6 co-sm-6"
                  label="Date"
                  error={errors.date}
                  type="date"
                  name="date"
                  value={date}
                  onChange={this.handleChange}
                />

                <InputGroup  
                  classes="co-xl-6 co-lg-6 co-md-6 co-sm-6"
                  label="Time"
                  error={errors.time}
                  type="time"
                  name="time"
                  value={time}
                  onChange={this.handleChange}
                />

                <InputGroup  
                  classes="co-xl-12 co-lg-12 co-md-12 co-sm-12"
                  label="Capacity"
                  error={errors.capacity}
                  type="number"
                  name="capacity"
                  max="100"
                  placeholder="max: 100"
                  value={capacity}
                  onChange={this.handleChange}
                />

                <InputGroup  
                  classes="co-xl-12 co-lg-12 co-md-12 co-sm-12"
                  label="Takeoff Venue"
                  error={errors.takeOffVenue}
                  type="text"
                  name="takeOffVenue"
                  placeholder="where will your journey start?"
                  value={takeOffVenue}
                  onChange={this.handleChange}
                />

                <div className="co-xl-12 co-lg-12 co-md-12 co-sm-12 right-text">
                  <Button 
                    text="CLOSE" 
                    type="button" 
                    onClick={this.closeModal.bind(this)}
                    classes="button button-white close"
                  />
                  {" "}

                  <Button
                    text="CREATE OFFER"
                    type="submit"
                    style={ isLoading ? {opacity: 0.4, cursor: 'forbidden'} : null}
                    classes="button button-blue create" 
                    disabled={ isLoading ? true : false}
                  />
                </div>
              </div>
            </form>
          </div>
        </div>
      </Fragment>
    );
  }
}

CreateRideModal.defaultProps = {
  modalName: 'CreateRideModal'
}

CreateRideModal.propTypes = {
  isCreateRideModalOpen: PropTypes.bool.isRequired,
  closeModal: PropTypes.func.isRequired,
  createRide: PropTypes.func.isRequired,
  modalName: PropTypes.string.isRequired,
  succesMessge: PropTypes.string,
  history: PropTypes.objectOf(PropTypes.any),
  isLoading: PropTypes.bool.isRequired
}


const mapStateToProps = (state) => ({
  isCreateRideModalOpen: state.modal.isCreateRideModalOpen,
  isLoading: state.ride.isLoading,
  succesMessge: state.ride.successMessge
});

export default withRouter(connect(
  mapStateToProps,
  { closeModal, createRide }
)(CreateRideModal));
