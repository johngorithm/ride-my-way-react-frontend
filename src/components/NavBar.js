import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import PropTypes from 'prop-types'


import { openModal } from 'actions/modalActions';
import { logOut } from 'actions/authActions';
import logo from '../static/images/favicon.png';

export class NavBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isDropDownVisible: false,
      isMobileNavVisible: false
    };
  }

  changeLink(link, event){
    event.preventDefault();
    this.setState({
      isDropDownVisible: false,
      isMobileNavVisible: false
    });
    this.props.history.push(link);
  }


  toggleDropdown(e) {
    e.preventDefault();
    this.setState({
      isDropDownVisible: !this.state.isDropDownVisible
    })
  }

  toggleMobileNav(e) {
    e.preventDefault();
    this.setState({
      isMobileNavVisible: !this.state.isMobileNavVisible
    })
  }

  openCreateRideModal(e) {
    e.preventDefault();
    this.setState({
      isDropDownVisible: false,
      isMobileNavVisible: false
    });
    this.props.openModal('CreateRideModal');
  }

  handleLogOut(e) {
    e.preventDefault();
    this.setState({
      isDropDownVisible: false,
      isMobileNavVisible: false
    });
    this.props.logOut().then(() => {
      this.props.history.push('/');
    });
  }

  render() {
    if (!this.props.isAuthenticated) {
      return (
        <nav className="navigation">
          <div className="navbar wrapper">
            <p className="navbar-toggle small" onClick={this.toggleMobileNav.bind(this)} >MENU</p>
            <ul className="nav-items nav-left">
              <li className="nav-item">
                <img src={logo} alt="logo" />
                <Link onClick={this.changeLink.bind(this, '/')}  to="/">RMW</Link>
              </li>
            </ul>
            <ul className="nav-items nav-right" style={this.state.isMobileNavVisible ? { display: 'block' } : null} >
              <li className="nav-item">
                <Link onClick={this.changeLink.bind(this, '/login')}  to="/login">LOGIN</Link>
              </li>
              <li className="nav-item">
                <Link onClick={this.changeLink.bind(this, '/register')}  to="/register">REGISTER</Link>
              </li>
            </ul>
          </div>
        </nav>
      );
    } else {
      return (
        <nav className="navigation">
          <div className="navbar wrapper">
            <p id="toggle-mobile-nav" className="navbar-toggle small" onClick={this.toggleMobileNav.bind(this)}>MENU</p>
            <ul className="nav-items nav-left">
              <li className="nav-item">
                <img src={logo} alt="logo" /> <Link onClick={this.changeLink.bind(this, '/')}  to="/">RMW </Link>
              </li>
            </ul>
            <ul className="nav-items nav-right" style={this.state.isMobileNavVisible ? { display: 'block' } : null}>
              <li className="nav-item">
                <Link onClick={this.changeLink.bind(this, '/home')}  to="/home">Dashboard</Link>
              </li>
              <li id="offer-ride-btn" className="nav-item">
                <Link id="open-create-ride-modal" to="#" onClick={this.openCreateRideModal.bind(this)}>Offer A Ride</Link>
              </li>
              <li id="dropdown-nav" className="nav-item">
                <Link id="dropdown-toggle" to="#" onClick={this.toggleDropdown.bind(this)}>
                  <span>{this.props.user.username}</span> <i className="fas fa-chevron-down" />
                </Link>
                {this.state.isDropDownVisible ? (
                  <div className="droppy">
                    <span className="pointer" />
                    <div className="dropdown-nav">
                      <Link onClick={this.changeLink.bind(this, '/profile')} id="first" to="/profile">
                        Profile
                    </Link>
                      <Link onClick={this.changeLink.bind(this, '/notification')} to="/notifications">Notifications</Link>
                      <Link id="logout" to="#" onClick={this.handleLogOut.bind(this)}>
                        Logout
                    </Link>
                    </div>
                  </div>
                ) : null }

              </li>
            </ul>
          </div>
        </nav>
      )
    }
  }
}

NavBar.propTypes = {
  openModal: PropTypes.func.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  user: PropTypes.object,
  logOut: PropTypes.func.isRequired,
  history: PropTypes.objectOf(PropTypes.any)
};

const mapStateToProps = (state) => ({
  isAuthenticated: state.auth.isAuthenticated,
  user: state.auth.user
})

export default withRouter(connect(
  mapStateToProps,
  { openModal, logOut }
)(NavBar));
