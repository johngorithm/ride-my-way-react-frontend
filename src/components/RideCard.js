import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { openModal } from 'actions/modalActions'
import { addRide } from 'actions/rideActions'

class RideCard extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  handleRideView() {
    this.props.addRide(this.props.ride);
    this.props.openModal('RideDetailModal');
  }

  convertTimeTo12HoursFormat = (time)  => {
    time = time.toString().match (/^([01]\d|2[0-3])(:)([0-5]\d)(:[0-5]\d)?$/) || [time];
  
    if (time.length > 1) { 
      time = time.slice (1);
      time[5] = +time[0] < 12 ? ' AM' : ' PM';
      time[0] = +time[0] % 12 || 12;
    }
    return time.join ('');
  }

  render() {
    const { time, date, destination, capacity, space_occupied } = this.props.ride
    return (
      <Fragment>
        <div className="co-xl-3 co-lg-4 co-md-6 co-sm-6">
          <div className="tile">
            <div className="tile-heading center-text">
              <p className="${(ownership ? 'tag' : '')}"></p>
              <h4>RIDE TO<br></br><span>{destination}</span></h4>
            </div>

            <div className="tile-body">
              <div className="row">
                <div className="co-xl-6 co-lg-6 co-md-6 co-sm-6 co-xs-6 center-text border">
                  <p className="data-heading small">Date</p>
                  <p className="small info">{(new Date(date)).toDateString().slice(4, 15)}</p>
                </div>
                <div className="co-xl-6 co-lg-6 co-md-6 co-sm-6 co-xs-6 center-text">
                  <p className="data-heading small">Time</p>
                  <p className="small info">{this.convertTimeTo12HoursFormat(time.slice(0, 5))}</p>
                </div>
              </div>
            </div>
            <div className="tile-body not-first left-text">
              <p className="left"><span className="badge">{capacity}</span>capacity</p>
              <p className="right"><span className="badge">{space_occupied}</span>occupied</p>
            </div>

            <div className="tile-footer center-text">
              <button onClick={this.handleRideView.bind(this)} data-ownership="{(ownership === 'yours') ? 'true' : 'false'}" data-ride='${JSON.stringify(ride)}' className="button button-blue view">VIEW</button>
            </div>
          </div>
        </div>
      </Fragment>
    );
  }
}

RideCard.propTypes = {
  openModal: PropTypes.func.isRequired,
  ride: PropTypes.object.isRequired,
  addRide: PropTypes.func.isRequired
};


export default connect(
  null,
  { openModal, addRide }
)(RideCard);
