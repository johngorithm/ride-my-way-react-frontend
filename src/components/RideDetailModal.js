import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { closeModal } from 'actions/modalActions';

class RideDetailModal extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <Fragment>
        <div className="modal" id="detail-modal">
          <div className="modal-content animate">
            <div className="tile">
              <div className="tile-heading center-text">
                <h4>
                  RIDE TO <span>{this.props.ride.destination}</span>
                </h4>
                <p className="smaller">
                  <strong>from: </strong>
                  <span>{this.props.ride.takeOffVenue}</span>
                </p>
                <span className="message smaller" />
              </div>

              <div className="tile-body">
                <div className="row">
                  <div className="co-xl-6 co-lg-6 co-md-6 co-sm-6 co-xs-6 center-text border">
                    <p className="data-heading small">Date</p>
                    <p className="small date">{this.props.ride.date}</p>
                  </div>
                  <div className="co-xl-6 co-lg-6 co-md-6 co-sm-6 co-xs-6 center-text">
                    <p className="data-heading small">Time</p>
                    <p className="small time">{this.props.ride.time}</p>
                  </div>
                </div>
              </div>

              <div id="capacity-info" className="tile-body not-first center-text">
                <p className="left">
                  capacity<span className="badge" id="capacity">{this.props.ride.capacity}</span>

                </p>
                <p className="right">
                  occupied<span className="badge" id="space-occupied"> {this.props.ride.occupied}</span>
                </p>
              </div>

              <div className="tile-body driver center-text padded">
                <p className="data-heading small">Driver</p>
                <p className="small driver">{this.props.ride.driver}</p>
              </div>

              <div className="tile-footer center-text">
                <button onClick={() => this.props.closeModal('RideDetailModal')} className="button button-blue close">CLOSE</button>
                <button
                  //onClick="joinRide(this)"
                  className="button button-white join"
                >
                  JOIN
                </button>
              </div>
            </div>
          </div>
        </div>
      </Fragment>
    );
  }
}

RideDetailModal.propTypes = {
  closeModal: PropTypes.func.isRequired,
  ride: PropTypes.object.isRequired
};

const mapStateToProps = (state) => ({
  ride: state.ride.ride
})

export default connect(
  mapStateToProps,
  { closeModal }
)(RideDetailModal);
