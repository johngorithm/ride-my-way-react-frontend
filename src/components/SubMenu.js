import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';


class SubMenu extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <Fragment>
        <div className="submenu row">
          {/**<!--SUBMENU BAR STARTS-->**/}
          <Link
            to="/home"
            className="link active co-xl-6 co-lg-6 co-md-6 co-sm-6 co-xs-6 center-text border"
          >
            RIDE OFFERS
        </Link>
          <Link
            to="/requests"
            className="co-xl-6 co-lg-6 co-md-6 co-sm-6 co-xs-6 center-text"
          >
            MY REQUESTS
        </Link>
        </div>
      </Fragment>
    );
  }
}



export default SubMenu;
