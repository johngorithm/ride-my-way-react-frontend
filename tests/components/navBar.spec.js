import React from 'react';
import { shallow } from 'enzyme';


import { NavBar } from 'components/NavBar';

const mockFn = jest.fn();

const props = {
  isAuthenticated: false,
  openModal: mockFn,
  logOut: jest.fn(() => Promise.resolve()),
  history: { push: jest.fn() },
  handleLogOut: mockFn,
  user: {
    user_id: 1,
    username: 'johngorithm',
    firstname: 'John'
  }
}
describe('NAVBAR TEST', () => {
  const wrapper = shallow(<NavBar {...props} />);

  test('should render without crashing', () => {
    expect(wrapper.exists()).toBe(true);
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('shows mobile nav', () => {
    wrapper.setState({
      isMobileNavVisible: true
    })
    expect(wrapper.find('[to="/login"]').exists()).toBe(true);
  });

  test('shows logged in nav', () => {
    wrapper.setProps({
      isAuthenticated: true
    })
    expect(wrapper.find('#offer-ride-btn').exists()).toBe(true);
  });

  test('shows dropdown user nav', () => {
    wrapper.setProps({
      isAuthenticated: true
    })
    wrapper.setState({
      isDropDownVisible: true
    })
    expect(wrapper.find('.droppy').exists()).toBe(true);
  });
  
  test('logs out a user', () => {
    wrapper.setProps({
      isAuthenticated: true
    })
    wrapper.setState({
      isDropDownVisible: true
    })
    expect(wrapper.find('#logout').exists()).toBe(true);
    const event = {
      preventDefault: mockFn
    };
    wrapper.instance().handleLogOut = mockFn;
    wrapper.update();

    wrapper.find('#logout').simulate('click', event);
    expect(wrapper.instance().props.logOut).toHaveBeenCalled()
  });

  test('opens create-ride-modal', () => {
    wrapper.setProps({
      isAuthenticated: true
    })

    expect(wrapper.find('#open-create-ride-modal').exists()).toBe(true);
    const event = {
      preventDefault: mockFn
    };

    wrapper.find('#open-create-ride-modal').simulate('click', event);
    expect(wrapper.instance().props.openModal).toHaveBeenCalledWith('CreateRideModal');
  });

  test('toggles dropdown nav', () => {
    wrapper.setProps({
      isAuthenticated: true
    })
    wrapper.setState({
      isDropDownVisible: true
    })
    expect(wrapper.find('#dropdown-toggle').exists()).toBe(true);
    const event = {
      preventDefault: mockFn
    };

    wrapper.find('#dropdown-toggle').simulate('click', event);
    expect(wrapper.instance().state.isDropDownVisible).toBe(false);
  });

  test('toggles mobile dropdown nav', () => {
    wrapper.setProps({
      isAuthenticated: true
    })
    wrapper.setState({
      isMobileNavVisible: true
    })
    expect(wrapper.find('#toggle-mobile-nav').exists()).toBe(true);
    const event = {
      preventDefault: mockFn
    };

    wrapper.find('#toggle-mobile-nav').simulate('click', event);
    expect(wrapper.instance().state.isMobileNavVisible).toBe(false);
  });

  test('switches between navs', () => {
    wrapper.setProps({
      isAuthenticated: false
    })

    // expect(wrapper.find('#toggle-mobile-nav').exists()).toBe(true);
    const event = {
      preventDefault: mockFn
    };

    wrapper.instance().changeLink = mockFn;
    wrapper.update();

    wrapper.find('[to="/login"]').simulate('click', event);
    expect(wrapper.instance().changeLink).toHaveBeenCalled();
  });
})