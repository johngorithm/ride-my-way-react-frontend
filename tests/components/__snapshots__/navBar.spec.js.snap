// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NAVBAR TEST should match snapshot 1`] = `
<nav
  className="navigation"
>
  <div
    className="navbar wrapper"
  >
    <p
      className="navbar-toggle small"
      onClick={[Function]}
    >
      MENU
    </p>
    <ul
      className="nav-items nav-left"
    >
      <li
        className="nav-item"
      >
        <img
          alt="logo"
          src="test-file-stub"
        />
        <Link
          onClick={[Function]}
          replace={false}
          to="/"
        >
          RMW
        </Link>
      </li>
    </ul>
    <ul
      className="nav-items nav-right"
      style={null}
    >
      <li
        className="nav-item"
      >
        <Link
          onClick={[Function]}
          replace={false}
          to="/login"
        >
          LOGIN
        </Link>
      </li>
      <li
        className="nav-item"
      >
        <Link
          onClick={[Function]}
          replace={false}
          to="/register"
        >
          REGISTER
        </Link>
      </li>
    </ul>
  </div>
</nav>
`;
